toolchain/lower/function_context.h

```diff
@@ -139,6 +146,45 @@ class FunctionContext {
   auto FinishInit(SemIR::TypeId type_id, SemIR::InstId dest_id,
                   SemIR::InstId source_id) -> void;
 
+  auto AddCallToCurrentFingerprint(SemIR::FunctionId function_id,
+                                   SemIR::SpecificId specific_id) {
+    if (!function_fingerprint_) {
+      return;
+    }
+    RawStringOstream os;
+    // TODO: Replace index with info that is translation unit independent.
+    // Using a string that includes the function_id string and the index to
+    // avoid possible collisions. This needs revisiting.
+    os << "function_id" << function_id.index;
+    current_fingerprint_.function_common_fingerprint.update(os.TakeStr());
+    // TODO: Replace index with info that is translation unit independent.
+    if (specific_id.has_value()) {
+      current_fingerprint_.function_specific_fingerprint.update(
+          specific_id.index);
+      function_fingerprint_->calls.push_back(specific_id);
+    }
+  }
```

Comment:
These function bodies seem a little large for the header, making it more difficult to read the class API. Had you considered moving the bodies out-of-line to the cpp file?